module/x/gravity/keeper/batch.go

```diff
@@ -88,6 +88,22 @@ func (k Keeper) batchTxExecuted(ctx sdk.Context, tokenContract common.Address, n
 		if (btx.BatchNonce < batchTx.BatchNonce) && (batchTx.TokenContract == tokenContract.Hex()) {
 			k.CancelBatchTx(ctx, tokenContract, btx.BatchNonce)
 		}
+
+		for _, tx := range btx.Transactions {
```

Comment:
Potential performance issue here when the transaction size is huge?

Reply:
this event exists on the gravity.sol smart contract side. Blockscout can also index this events natively (since its erc20 transfer)

What is the use case of this event (which is more an aknowledgement)?

Reply:
Yeah, should it be out of the for loop?
And is the event in the ethereum side enough? https://github.com/PeggyJV/gravity-bridge/blob/main/solidity/contracts/Gravity.sol#L391